### PR TITLE
SOASTA 2015 Contributions

### PR DESCRIPTION
**Pull request by nicjansma**
This is a master pull request of all of SOASTA's commits to boomerang since December 2014.

Some of the primary features are:

* Single Page App (SPA) support for AngularJS, Ember.js and Backbone.js
* Updated `restiming.js` plugin for compressed ResourceTimings
* Full end-to-end (E2E) integration test suite
* Revamped build process using Grunt
* `.editorconfig` and improved `.eslintrc` with many lint fixes (note one commit that made indents consistent causes many diffs to look complex)

### CODE REVIEW

File: boomerang.js [hunk 1]
```diff
@@ -369,780 +363,986 @@ boomr = {
...
+			};
+		},
+
+		responseEnd: function(name, t_start, data) {
+			if (typeof name === "object" && name.url) {
+				impl.fireEvent("xhr_load", name);
 			}
-		}
+			else {
+				BOOMR.addVar("xhr.pg", name);
```
Comment (andreas-marschke) [inline boomerang.js]: xhr.pg required in OSS Version?

File: boomerang.js [hunk 2]
```diff
@@ -369,780 +363,986 @@ boomr = {
...
+					"name": "xhr_" + name,
+					"data": data
+				});
+			}
+		},
+
+		/**
+		 * Undo XMLHttpRequest instrumentation and reset the original
+		 */
+		uninstrumentXHR: function() {
```
Comment (andreas-marschke) [inline boomerang.js]: if instrumentXHR and uninstrumentXHR are implemented in auto_xhr.js why is this here?

File: boomerang.js (hunk 1 above)
Comment (bluesmoon) [inline boomerang.js]: no, it should not be in there.

File: boomerang.js (hunk 2 above)
Comment (bluesmoon) [inline boomerang.js]: just an API stub, so that it can be documented.
